cmd/otel-allocator/internal/watcher/promOperator.go

```diff
@@ -53,7 +53,30 @@ func NewPrometheusCRWatcher(ctx context.Context, logger logr.Logger, cfg allocat
 		return nil, err
 	}
 
-	factory := informers.NewMonitoringInformerFactories(map[string]struct{}{v1.NamespaceAll: {}}, map[string]struct{}{}, mClient, allocatorconfig.DefaultResyncTime, nil) //TODO decide what strategy to use regarding namespaces
+	if len(cfg.PrometheusCR.AllowNamespaces) != 0 && len(cfg.PrometheusCR.DenyNamespaces) != 0 {
+		return nil, fmt.Errorf("both allow and deny namespaces are set, only one should be set")
+	}
+
+	allowList := map[string]struct{}{}
```

Comment:
minor suggestion, take it or leave it... Can we move the construction of these lists to a method on the PrometheusCR object?